metrics/density/footprint_data.md

```diff
@@ -0,0 +1,104 @@
+# Footprint data script details
+
+The `footprint_data.sh` test within this directory can be configured with a number
+of environment variables.
+
+Details of the configurable variables are listed below, along with a small example
+invocation script.
+
+## Variables
+The script action can be modified by setting a number of environment variables.
+
+These variables take effect generally in two ways. Some variables affect how the
+payload is executed. The `RUNTIME` and `PAYLOAD` arguments directly affect the
+payload execution via this line in the script:
+
+`docker run -tid --runtime=$RUNTIME $PAYLOAD_RUNTIME_ARGS $PAYLOAD $PAYLOAD_ARGS`
+
+Other settings affect how memory footprint is measured and the test termination
+conditions.
+
+| Variable | Function
+| -------- | --------
+| PAYLOAD | The docker image to run
+| PAYLOAD_ARGS | Any arguments passed into the docker image
+| PAYLOAD_RUNTIME_ARGS | Any extra arguments passed into the docker `run` command
+| PAYLOAD_SLEEP | Seconds to sleep between launch and measurement, to allow settling
+| MAX_NUM_CONTAINERS | The maximum number of containers to run before terminating
+| MAX_MEMORY_CONSUMED | The maximum amount of memory to be consumed before terminating
+| MIN_MEMORY_FREE | The minimum amount of memory allowed to be free before terminating
+| CC_QEMU_PATH | The path to the Clear Containers QEMU binary (for `smem` measurements)
+| DOCKERD_PATH | The path to the Docker `dockerd` binary (for `smem` measurements)
+| DUMP_CACHES | A flag to note if the system caches should be dumped before capturing stats
+| DATAFILE | Can be set to over-ride the default CSV results filename
+| DATAFILE | Can be set to over-ride the default CSV results filename
+
+## Output files
+The names of the CSV files generated by the test are dictated by some of the parameters
+the test is utilising. The default filename is defined by the expansion of:
+`${results-${RUNTIME}-${PAYLOAD}.csv}`
```

Comment:
It might be a useful feature to have a script option (`-n` / `--dry-run` or something) that will just display the name of the file that would be generated?